src/hotspot/share/gc/shared/gcVMOperations.cpp

```diff
@@ -46,6 +46,32 @@
 #include "gc/g1/g1Policy.hpp"
 #endif // INCLUDE_G1GC
 
+bool VM_GC_Sync_Operation::doit_prologue() {
+  Heap_lock->lock();
+
+  // Check invocations
+  if (skip_operation()) {
+    // skip collection
+    Heap_lock->unlock();
+    _prologue_succeeded = false;
+  } else {
+    _prologue_succeeded = true;
+  }
+  return _prologue_succeeded;
+}
```

Comment:
This invocation checking doesn't seem right at this level.  That is, skip_operation and prologue_succeeded all seem to me to have nothing to do with syncing, instead belonging to the VM_GC_Operation level and should remain there.